src/compiler/utilities.ts

```diff
@@ -1779,6 +1771,31 @@ namespace ts {
         }
     }
 
+    export function isPartOfTypeQuery(node: Node) {
+        while (node.kind === SyntaxKind.QualifiedName || node.kind === SyntaxKind.Identifier) {

```

Comment:
As a minor optimization, I think that `QualifiedName`s are always parts of types, but don't take my word for it maybe.

Reply:
I think you’re right.

Reply:
This would be true if it weren't for `import foo = a.b.c`! There's `PropertyAccessEntityName` but I'm not sure if that helps.

Reply:
But, this util currently has a reasonable name and a reasonable chance of reuse. Not sure an optimization here is worth the hit to the understandability of what we’re checking.